#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/appearances/protos/MattePaint.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0944
  position -2.1937665412347754e-05 -2.193772744394873e-05 2.2416791848467192
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  wallHeight 0.15
  wallAppearance MattePaint {
  }
}
DEF YellowCylinder Solid {
  translation -0.5 0.5 0.1492510966575714
  rotation -2.8533290763486366e-17 2.109308606214159e-17 1 1.5707963267948961
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.38
        radius 0.08
      }
    }
  ]
  name "Yellow Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  recognitionColors [
    1 1 0
  ]
}
DEF RedCylinder Solid {
  translation 0.5 0.5 0.1492510966575714
  rotation -4.5712411693179405e-17 -8.229072391572502e-18 1 1.5707963267948961
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.38
        radius 0.08
      }
    }
  ]
  name "Red Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 1 0 0
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  recognitionColors [
    1 0 0
  ]
}
DEF GreenCylinder Solid {
  translation -0.5 -0.5 0.1492510966575714
  rotation -1.7732517558419064e-17 2.5172722259764433e-17 1 1.5707963267948961
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.38
        radius 0.08
      }
    }
  ]
  name "Green Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 1 0
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  recognitionColors [
    0 1 0
  ]
}
DEF BlueCylinder Solid {
  translation 0.5 -0.5 0.1492510966575714
  rotation 2.1143250656105032e-17 1.047761932461218e-17 1 1.5707963267948961
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.38
        radius 0.08
      }
    }
  ]
  name "Blue Landmark"
  boundingObject DEF landmark Shape {
    appearance PBRAppearance {
      baseColor 0 0 1
      roughness 1
      metalness 0
    }
    geometry Cylinder {
      height 0.3
      radius 0.045
    }
  }
  physics DEF landmarkPH Physics {
  }
  recognitionColors [
    0 0 1
  ]
}
DEF epuck E-puck {
  hidden position_0_0 1.7279513174669197
  hidden position_0_1 -1.7279513171250571
  hidden linearVelocity_0 -3.4721789888047846e-06 -1.7687620156982315e-05 -1.3652104750389315e-09
  hidden angularVelocity_0 8.96491595092209e-05 -0.00023828048648516335 -0.7023125774701137
  hidden rotation_1 0 1 0 1.7279513174669197
  hidden linearVelocity_1 0.017060000874836473 0.006540776040283889 -2.713003599552806e-10
  hidden angularVelocity_1 -0.3585678963923941 0.9332952313041135 -0.7040648059710135
  hidden rotation_2 0 -1 0 1.7279513171250571
  hidden linearVelocity_2 -0.017069548193283688 -0.006576552280593232 1.5280909870798618e-10
  hidden angularVelocity_2 0.3587475314593336 -0.9337716557708003 -0.704067816549596
  translation 0.37499981190873954 -0.37504635301117256 -6.50966149626387e-05
  rotation 4.17144996868308e-05 -0.00022099603367326624 0.9999999747103264 0.35529117566180224
  controller "Lab5_Task2"
  supervisor TRUE
  turretSlot [
    DEF Front DistanceSensor {
      translation 0.045 0 -0.011
      name "front distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Right DistanceSensor {
      translation 0.01 -0.035 -0.011
      rotation 0 0 1 -1.5707853071795865
      name "right distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Rear DistanceSensor {
      translation -0.025 0 -0.011
      rotation 0 0 1 3.1415
      name "rear distance sensor"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF Left DistanceSensor {
      translation 0.01 0.035 -0.011
      rotation 0 0 1 1.5708
      name "left distance sensor"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    InertialUnit {
      rotation -1 0 0 -5.307179586466759e-06
    }
    Camera {
      translation 0.043 0 -0.015
      name "front camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation 0 0.043 -0.015
      rotation 0 0 1 1.57
      name "right camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation -0.043 0 -0.015
      rotation 0 0 1 3.14
      name "rear camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
    Camera {
      translation 0 -0.043 -0.015
      rotation 0 0 1 -1.57
      name "left camera"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
